pkg/datapath/loader/tc.go

```diff
@@ -48,20 +56,28 @@ func attachSKBProgram(device netlink.Link, prog *ebpf.Program, progName, bpffsDi
 	}
 
 	// Legacy tc attached, make sure tcx is detached in case of downgrade.
+	// netkit can only be used in combination with tcx, but never legacy tc,
+	// hence for netkit detaching here would be irrelevant.
 	if err := detachTCX(bpffsDir, progName); err != nil {
 		return fmt.Errorf("tcx cleanup after attaching legacy tc program %s: %w", progName, err)
 	}
 
 	return nil
 }
 
-// detachSKBProgram attempts to remove an existing tcx and legacy tc link with
-// the given properties. Always attempts to remove both tcx and tc attachments.
+// detachSKBProgram attempts to remove an existing tcx, netkit and legacy tc link
+// with the given properties. Always attempts to remove all three attachments.
 func detachSKBProgram(device netlink.Link, progName, bpffsDir string, parent uint32) error {
 	if err := detachTCX(bpffsDir, progName); err != nil {
 		return err
 	}
 
+	if device.Type() == "netkit" {
+		if err := detachNetkit(bpffsDir, progName); err != nil {
```

Comment:
I think `detachTCX` might already have removed the link at this point because nothing in the bpffs dir is netkit specific? And if it fails, even though we are removing links attached to netkit devices the error strings will mention tcx.
I think we should merge `detachTCX` and `detachNetkit` into one `bpf.RemoveLink` function or something similar, and use it in both places. Essentially `detachTCX` and `detachNetkit` only differ wrt to error strings being returned which we can add when calling `bpf.RemoveLink`. WDYT?

Reply:
Good point, yes, I'll merge the two.